File: Codes/0080-remove-duplicates-from-sorted-array-ii_20250209.java
/*
 * @Author: LetMeFly
 * @Date: 2025-02-09 08:53:42
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-02-09 08:58:07
 */
class Solution {
    public int removeDuplicates(int[] nums) {
        int slow = 1;
        for (int fast = 2; fast < nums.length; fast++) {
            if (nums[fast] != nums[slow - 1]) {
                nums[++slow] = nums[fast];
            }
        }
        return Math.min(slow + 1, nums.length);
    }
}

File: Codes/0080-remove-duplicates-from-sorted-array-ii_20250209.go
/*
 * @Author: LetMeFly
 * @Date: 2025-02-09 08:53:44
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-02-09 09:00:09
 */
package main

func removeDuplicates(nums []int) int {
    slow := 1
    for fast := 2; fast < len(nums); fast++ {
        if nums[fast] != nums[slow - 1] {
            slow++
            nums[slow] = nums[fast]
        }
    }
    return min(slow + 1, len(nums))
}